kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/utils/TestJsonMetadataDomain.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.utils;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.delta.kernel.internal.metadatadomain.JsonMetadataDomain;
+
+import java.util.Optional;
+
+/**
+ * A test implementation of {@link JsonMetadataDomain} for testing purposes. It has two Optional
+ * fields and one primitive field.
+ */
+public class TestJsonMetadataDomain extends JsonMetadataDomain {
```

Comment:
can this be reduced to `case` class in scala in the same DomainMetadataSuite.scala?

Reply:
I looked into this, and one issue is `ObjectMapper` in `JsonMetadataDomain.java` doesn’t have the `DefaultScalaModule` needed to handle Scala’s `Option`. But for sure we could add the dependency and explicitly register it if we really want this.

Also, IMO using a `TestJsonMetadataDomain` Java class might be better since it allows us to directly test serialization and deserialization of Java classes, which reflects real usages more accurately (e.g. it tests Java's `Optional` instead of Scala's `Option`).